src/main/java/org/hiero/sketch/dataset/api/PartialResult.java

```diff
@@ -0,0 +1,40 @@
+package org.hiero.sketch.dataset.api;
+
+import java.io.Serializable;
+
+/**
+ * A partial result always describes a DELTA between the previous partial result
+ * and the current one.
+ * @param <T>  Type of data in partial result.
+ */
+public class PartialResult<T> implements Serializable {
+    /**
+     * How much more has been deltaDone from the computation has been deltaDone
+     * since the previous partial result.  A number between 0 and 1.
+     */
+    public final double deltaDone;
+    /**
+     * Additional data computed since the previous partial result.
+     */
+    public final T deltaValue;
+
+    /**
+     * Creates a partial result.
+     * @param deltaDone  How much more has been done.  A number between 0 and 1.
+     * @param deltaValue Extra result produced.
+     */
+    public PartialResult(double deltaDone, final T deltaValue) {
+        if (deltaDone < 0)
```

Comment:
"deltaDone < 0" sounds like an exception or even something to check with an assert. Why assign deltaDone to 0 instead?